Subject: [PATCH 5/5] support payload post

Add not only the '@odata.id', but also the other payload into the newly added
member.

Also fixed a bug when the 1st member doesn't contain ending digits, it
fails into a infinite loop.

Signed-off-by: Nan Zhou <nanzhou@google.com>
---
 redfishMockupServer.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/redfishMockupServer.py b/redfishMockupServer.py
index b44ee21..2141880 100644
--- a/redfishMockupServer.py
+++ b/redfishMockupServer.py
@@ -145,16 +145,21 @@ class RfMockupServer(BaseHTTPRequestHandler):
         def add_new_member(self, payload, data_received):
             members = payload.get('Members')
             n = 1
-            pattern = re.sub(r'\d+$', '{id}', members[0].get('@odata.id').replace(self.path, '').strip('/')) if len(members) else 'Member{id}'
+            pattern = 'Member{id}'
+            if len(members) > 0:
+                member = members[0].get('@odata.id').replace(self.path, '').strip('/')
+                if re.search(r'\d+$', member) is not None:
+                    pattern = re.sub(r'\d+$', '{id}', member)
             newpath_id = data_received.get('Id', pattern.format(id=n))
             newpath = '/'.join([ self.path, newpath_id ])
             while newpath in [m.get('@odata.id') for m in members]:
                 n = n + 1
                 newpath_id = data_received.get('Id', pattern.format(id=n))
                 newpath = '/'.join([ self.path, newpath_id ])
-            members.append({'@odata.id': newpath})
+
             data_received['@odata.id'] = newpath
             data_received['Id'] = newpath_id
+            members.append(data_received)

             payload['Members'] = members
             payload['Members@odata.count'] = len(members)
--
2.31.1.498.g6c1eba8ee3d-goog
